src/sc/sidechain.cpp

```diff
@@ -783,108 +782,91 @@ void CSidechain::UpdateScFees(const CScCertificateView& certView, int blockHeigh
-            scFees_v2.emplace_back(ftScFee, mbtrScFee, blockHeight);
+        if (scFeeIt == scFees.end()) {
+            scFees.emplace_back(new Sidechain::ScFeeData_v2(ftScFee, mbtrScFee, blockHeight));
 
             // as for v1 sidechains
             const size_t max_size { static_cast<size_t>(maxSizeOfScFeesContainers) };
-            while (scFees_v2.size() > max_size)
+
+            while (scFees.size() > max_size)
             {
-                const auto& entry = scFees_v2.front();
-                LogPrint("sc", "%s():%d - popping f=%d, m=%d, h=%d from list, as scFees maxsize has been reached\n",
-                    __func__, __LINE__, entry.forwardTxScFee, entry.mbtrTxScFee, entry.submissionHeight);
-                scFees_v2.pop_front();
+                const auto& entry = scFees.front();
+                LogPrint("sc", "%s():%d - popping %s from list, as scFees maxsize has been reached\n",
+                    __func__, __LINE__, entry->ToString().c_str());
+                scFees.pop_front();
             }
 
             // Also purge all elements submitted within a block that now is too old to be considered
             const auto threshold{blockHeight - this->maxSizeOfScFeesContainers};
-            scFees_v2.remove_if([threshold](Sidechain::ScFeeData_v2 entry) {
-                const bool testCondition = (entry.submissionHeight <= threshold);
+            scFees.remove_if([threshold](std::shared_ptr<Sidechain::ScFeeData> entry) {
+                const std::shared_ptr<Sidechain::ScFeeData_v2> casted_entry = std::dynamic_pointer_cast<Sidechain::ScFeeData_v2>(entry);
+                const bool testCondition = (casted_entry->submissionHeight <= threshold);
                 if (testCondition)
-                    LogPrint("sc", "%s():%d - popping f=%d, m=%d, h=%d from list, as entry is too old (threshold height was %d)\n",
-                        __func__, __LINE__, entry.forwardTxScFee, entry.mbtrTxScFee, entry.submissionHeight, threshold);
+                    LogPrint("sc", "%s():%d - popping %s from list, as entry is too old (threshold height was %d)\n",
+                        __func__, __LINE__, casted_entry->ToString().c_str(), threshold);
                 return testCondition;
             });
+
         }
         // On the other hand, if we found a scFeeData element for the current height, we just update it with lower
         // ft and/or mbtr fee rates. All the checks on the container size and element age have been performed
         // at the moment of the element insertion, hence their are not required here
         else {
-            scFeeIt->forwardTxScFee = std::min(scFeeIt->forwardTxScFee, ftScFee);
-            scFeeIt->mbtrTxScFee = std::min(scFeeIt->mbtrTxScFee, mbtrScFee);
+            scFeeIt->get()->forwardTxScFee = std::min(scFeeIt->get()->forwardTxScFee, ftScFee);
+            scFeeIt->get()->mbtrTxScFee    = std::min(scFeeIt->get()->mbtrTxScFee, mbtrScFee);
```

Comment:
I suggest for better readability 
```suggestion
            auto scFeeEntry = scFeeIt->get();
            scFeeEntry->forwardTxScFee = std::min(scFeeEntry->forwardTxScFee, ftScFee);
            scFeeEntry->mbtrTxScFee    = std::min(scFeeEntry->mbtrTxScFee, mbtrScFee);```

Reply:
Done